### PR TITLE
Event section-wip

### PR DESCRIPTION
**Pull request by AlexNechayev**
init of event section (work in progress)

### CODE REVIEW

File: pages/events/[...slug].vue [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    .only(["_path", "title", "description"])
+    .sort({ date: 1 })
+    .findSurround(path);
+  return {
+    event: await event,
+    surround: await surround,
+  };
+});
+
+// destrucure `prev` and `next` value from data
```
Comment (dimshik100) [inline pages/events/[...slug].vue]: Please pay attention to correct grammar.
``` destrucure -> destructure ```

File: components/Tags.vue [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+}
+.event-tags {
+  @apply grid grid-cols-3 transition-all max-h-0 overflow-hidden gap-2 ;
+  @apply select-none;
+  @apply md:max-w-0 md:max-h-max md:grid-cols-4;
+  /* @apply transition-all max-w-0 max-h-10 overflow-hidden flex flex-row justify-center content-center; */
+}
+.event-tags .tag {
+  @apply rounded-md bg-slate-100 p-2 py-1 text-sm text-slate-700;
+  @apply !py-0.5 transition-all hover:-translate-y-0.5;
```
Comment (dimshik100) [inline components/Tags.vue]: Check if this effect can be achieved without using `!important`.
This effect is causing the top of the tag to disappear - goes outside of the bounding box.

File: pages/events/index.vue [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    const searchParams = new URLSearchParams(url.search);
+
+    // Remove all search parameters
+    searchParams.delete(key);
+    const newUrl = url.origin + url.pathname + searchParams.toString();
+    // Replace the current URL with the new URL
+    history.replaceState(null, '', newUrl);
+
+  }
+}
```
Comment (dimshik100) [inline pages/events/index.vue]: Consider moving all of this functionality to an external file, as this is a helper stuff that we might want to use in multiple pages.

Please fix the `%2C` the should be a regular `,` in the URL.
Check if this helps -> https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURI

File: assets/css/main.css [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+h5,
+h6 {
+  @apply font-cabinetGrotesk;
+}
+
+.page-heading{
+    @apply flex justify-center;
+  }
+
+.cta {
```
Comment (dimshik100) [inline assets/css/main.css]: Please change this class name to something more specific.
Consider moving it to a more relevant location - as tags are only relevant in the main events page and not on all the website.

File: assets/css/main.css [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+h1,
+h2,
+h3,
+h4,
+h5,
+h6 {
+  @apply font-cabinetGrotesk;
+}
+
+.page-heading{
```
Comment (dimshik100) [inline assets/css/main.css]: Looks like this should also be inside event page

File: components/SiteHeader.vue [hunk 6]
```diff
@@ -0,0 +1,33 @@
+<!-- ./components/SiteHeader.vue -->
+
+<template>
+    <header class="site-header">
+      <div class="wrapper">
+        <NuxtLink to="/" class="no-underline flex">
+            <img class="w-8 mr-2" src="/logo/logo.svg" alt="Vue.js Israel Community Logo"> Vue.js Israel Community
```
Comment (dimshik100) [inline components/SiteHeader.vue]: Please rename `logo.svg` to `Vue.js Israel community - logo.svg`

File: components/Tags.vue [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+<style scoped>
+.tag-list {
+  @apply flex flex-col gap-2 p-4 rounded-lg ;
+  @apply md:items-center md:flex-row;
+}
+.event-tags {
+  @apply grid grid-cols-3 transition-all max-h-0 overflow-hidden gap-2 ;
+  @apply select-none;
+  @apply md:max-w-0 md:max-h-max md:grid-cols-4;
+  /* @apply transition-all max-w-0 max-h-10 overflow-hidden flex flex-row justify-center content-center; */
```
Comment (dimshik100) [inline components/Tags.vue]: Please delete if not in use.

File: components/events/EventCard.vue [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    weekday: "short",
+    year: "numeric",
+    month: "short",
+    day: "numeric",
+    hour: "numeric",
+    hour12: true,
+    minute: "numeric",
+    timeZoneName: "shortGeneric",
+    timeZone: "Asia/Jerusalem",
+  };
```
Comment (dimshik100) [inline components/events/EventCard.vue]: Please remove if not in use.

File: content/events/data-management-with-vue.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
+---
+title: "Data Management with Vue"
+description: "Vue Israel Meetup"
+date: "02 May 2023 18:00"
+location: "Semperis"
+tags: [Web, Development, Frameworks, Frontend, JavaScript, Vue.js]
```
Comment (dimshik100) [inline content/events/data-management-with-vue.md]: Please change this to `event-tags` as `tags` is a reserved key (apparently).
And fix(remove) the unneeded functionality related to `:static-tags`.

File: nuxt.config.ts [hunk 10]
```diff
@@ -1,4 +1,18 @@
...
+  content: {
+    // https://content.nuxtjs.org/api/configuration
+    highlight: {
+      theme: {
+        // Default theme (same as single string)
+        default: 'material-palenight',
+        // Theme used if `html.dark`
+        dark: 'github-dark',
+      }
+    }
```
Comment (dimshik100) [inline nuxt.config.ts]: This is not in use yet.
Please remove it for now.
We will add the necessary theme stuff when we get to light/dark modes.

File: public/images/img1.png
Comment (dimshik100) [inline public/images/img1.png:1]: Please update all header images to be in a convention name that looks similar to this `Vue.js Israel meetup - Semperis - 02.05.23.png`
Also, make sure that those images have the relevant ALT where they are used.

File: pages/events/index.vue [hunk 11]
```diff
@@ -0,0 +1,145 @@
+<script setup>
+useHead({
+  title: "All Events",
```
Comment (dimshik100) [inline pages/events/index.vue]: Please change to `Vue.js Israel Events`

File: pages/events/[...slug].vue [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+    .sort({ date: 1 })
+    .findSurround(path);
+  return {
+    event: await event,
+    surround: await surround,
+  };
+});
+
+// destructure `prev` and `next` value from data
+const [prev, next] = data.value.surround;
```
Comment (dimshik100) [inline pages/events/[...slug].vue]: Please remove all `surround` related code as we are not going to use it.

File: pages/events/[...slug].vue [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+// set the meta
+useHead({
+  title: `Vue.js Israel | ${data.value.event.title}`,
+  meta: [
+    { name: "description", content: data.value.event.description },
+    {
+      hid: "og:image",
+      property: "og:image",
+      content: `https://site.com/${data.value.event.img}`,
+    },
```
Comment (dimshik100) [inline pages/events/[...slug].vue]: Please remove this section, as we will address all SEO later.

**Review by dimshik100 — CHANGES_REQUESTED** [review]

**Comment by AlexNechayev** [comment]
@dimshik100 Hey,
I did all the requested changes.

**Review by dimshik100 — APPROVED** [review]

**Pull request merged**